### PR TITLE
Add config option to set default value for optional fields to `None` 

### LINKED ISSUE #152: Config for optional properties need to be explicitly set as `None`
Today, by default we have pass `None` for optional properties when building a pydantic object. This can either be viewed as tedious or viewed as a feature that will generate compile breaks when APIs change. 
Screen Shot 2023-01-03 at 11 14 41 AM

We should default this behavior to off but let it be configurable. The config should be similar to https://github.com/fern-api/fern-typescript/issues/57

### PR DESCRIPTION
**Pull request by GfxKai**
Closes #152 

- Add `require_optional_fields: bool = True` to `PydanticModelCustomConfig`

- If `False`, set default value for all optional fields to `None`

- Add snapshot test for `"require_optional_fields": False`

### CODE REVIEW

**Comment by dsinghvi** [comment]
@GfxKai you are crushing it, very impressed with your contributions

**Comment by dsinghvi** [comment]
Added to my list to review and will merge tonight/tomorrow

**Comment by GfxKai** [comment]
I guess this may need some doc updates too—currently there aren't any config options listed: https://buildwithfern.com/docs/compiler/generators#fern-pydantic-model

File: src/fern_python/generators/pydantic_model/fern_aware_pydantic_model.py [hunk 1]
```diff
@@ -246,14 +246,20 @@ def _create_pydantic_field(
         type_reference: ir_types.TypeReference,
         description: Optional[str] = None,
     ) -> PydanticField:
+        type_hint = self.get_type_hint_for_type_reference(type_reference)
+        default_value = (
+            AST.TypeHint.none()
```
Comment (GfxKai) [inline src/fern_python/generators/pydantic_model/fern_aware_pydantic_model.py:251]: @dsinghvi getting a mypy error here as `TypeHint` doesn't extend `Expression`.

Had a look but the best I could find is to pass it as a direct string: `AST.Expression("None")`—is there a better way of setting `None` here?

Comment (dsinghvi) [reply src/fern_python/generators/pydantic_model/fern_aware_pydantic_model.py]: Thats perfect

**Comment by GfxKai** [comment]
@dsinghvi can we roll an RC for this? Let me know if anything else needs doing 👍

**Comment by dsinghvi** [comment]
@GfxKai sorry for delay, merging now

**Pull request merged**